lib/fs.js

```diff
@@ -2070,11 +2074,43 @@ function writeAll(fd, isUserFd, buffer, offset, length, signal, callback) {
     } else {
       offset += written;
       length -= written;
-      writeAll(fd, isUserFd, buffer, offset, length, signal, callback);
+      writeAll(
+        fd, isUserFd, buffer, offset, length, signal, encoding, callback);
     }
   });
 }
 
+async function writeAllCustomIterable(
+  fd, isUserFd, buffer, offset, length, signal, encoding, callback) {
+  const result = await buffer.next();
+  if (result.done) {
+    fs.close(fd, callback);
+    return;
+  }
+  const resultValue = result.value.toString();
+  fs.write(fd, resultValue, undefined,
+           isArrayBufferView(buffer) ? resultValue.byteLength : encoding,
+           null, (writeErr, _) => {
+             if (writeErr) {
+               handleWriteAllErrorCallback(fd, isUserFd, writeErr, callback);
+             } else {
+               writeAll(fd, isUserFd, buffer, offset,
```

Comment:
maybe I'm missing something, but doesn't it make more sense to call `writeAllCustomIterable` here?

Reply:
In general, I think that there's a mistake here that also exists in the `fs/promises` version, where the code assumes that the whole chunk has been written but it might have only been partially written.